catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -93,6 +123,62 @@ public void init() {
 
     }
 
+    /**
+     * returns a Map of the parsed Accept-Encoding header according to RFC-2616-14.2
+     * </br>
+     * https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html
+     *
+     * @param header Accept-Encoding header to parse
+     * @return Map of {Encoding, Q value}
+     */
+    private Map<String, String> parseAcceptEncodings(String header) {
+        if (StringUtils.isEmpty(header)) {
+            return Collections.emptyMap();
+        }
+        String[] encodings = commaSpace.split(header.trim());
+        return Arrays.stream(encodings)
+                .map(semicolon::split)
+                .collect(Collectors.toMap(v -> v[0],
```

Comment:
I think Phil is asking how all of this might behave if someone were to provide an Accept-Encoding header with extremely large input, input with special characters that shouldn't be allowed, etc. If we get an Accept-Encoding header with 5000 characters in it, it is probably just malicious input, for example.